packages/workbox-strategies/src/CacheFirst.ts

```diff
@@ -89,23 +50,17 @@ class CacheFirst implements RouteHandlerObject {
       });
     }
 
-    let response = await cacheWrapper.match({
-      cacheName: this._cacheName,
-      request,
-      event,
-      matchOptions: this._matchOptions,
-      plugins: this._plugins,
-    });
+    let response = await handler.cacheMatch(request);
 
     let error;
     if (!response) {
       if (process.env.NODE_ENV !== 'production') {
         logs.push(
-            `No response found in the '${this._cacheName}' cache. ` +
-          `Will respond with a network request.`);
+            `No response found in the '${this.cacheName}' cache. ` +
```

Comment:
Something to think about that didn't occur to me when looking at this proposal initially: maybe this is an opportunity to revamp our log grouping logic?

Right now, there's an attempt at grouping being performed at the strategy level, by pushing everything into an array and logging each message at the end  in a group. But any messages that are generated by plugins will (still) end up outside of that group.

What if a plugin + strategy combo could all push messages into the same shared array, and those messages were all logged automatically at the end of strategy execution? (Maybe in a `finally` block to ensure that exceptions didn't swallow logging.) Would initializing `state.logs = []` before calling any plugins, and then having plugins push onto be sufficient for that sort of thing, or do we want to avoid "magic" properties on `state`?

Reply:
Yep, I'd actually been thinking about that as well. I figure we could have a logger plugin that all strategies share (or something like that). It'd make the logic in the strategy classes a lot easier to follow.

I was thinking of doing it as part of addressing this issue: https://github.com/GoogleChrome/workbox/issues/1858

Reply:
Moving the discussion to https://github.com/GoogleChrome/workbox/issues/1858 then.